### ISSUE TITLE
Allow descriptors represented in yaml

### ISSUE THREAD

**Issue by fjuniorr** [frictionlessdata/frictionless-r#66]
According to the specs all descriptors (eg. data package, table schema, etc) MUST be a JSON object. However, there are some [discussions](https://github.com/frictionlessdata/specs/issues/292) to support a YAML representation, and [frictionless-py already supports it](https://framework.frictionlessdata.io/docs/faq/#why-am-i-getting-the-error-cannot-extract-tabular-data-from-json).

This is a nice feature because it allows for a more readable and diffable documentation, especially for the `description` property in table schema which can make use of [YAML multiline strings](http://yaml-multiline.info/) for writing markdown without suffering the [pain of the JSON handling of newlines](https://stackoverflow.com/questions/42068/how-do-i-handle-newlines-in-json).

Compare

```yaml
fields:
- name: COD
  type: integer
  format: default
  description: |
    Código de dois dígitos que identificam o estado

    > Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  title: Código do Estado
- name: NOME
  type: string
  format: default
  title: Nome do Estado
- name: SIGLA
  type: string
  format: default
  title: Sigla do Estado
```

with

```json
{
  "fields": [
    {
      "name": "COD",
      "type": "integer",
      "format": "default",
      "description": "Código de dois dígitos que identificam o estado\n\n> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\n",
      "title": "Código do Estado"
    },
    {
      "name": "NOME",
      "type": "string",
      "format": "default",
      "title": "Nome do Estado"
    },
    {
      "name": "SIGLA",
      "type": "string",
      "format": "default",
      "title": "Sigla do Estado"
    }
  ]
}
```

Because some [data packages are going to mix both JSON and YAML](https://github.com/dados-mg/datapackage-reprex/tree/yaml-schema), it would be nice to be able to write

```R
library("frictionless")

dp <- read_package("https://raw.githubusercontent.com/dados-mg/datapackage-reprex/yaml-schema/datapackage.json")
df <- dp |> read_resource("estados")
```

which currently fails with

```
Error in parse_con(txt, bigint_as_char) : 
  lexical error: invalid string in json text.
                                       fields: - name: COD   type: int
                     (right here) ------^
```

**Comment by peterdesmet** [comment]
I understand how yaml can be more readable than json, but do we want to allow mixing json (`datapackage.json`) with yaml (`schema.yaml`)? That makes it harder for implementors. I'm inclined to only support yaml if it is in the specs.

**Comment by fjuniorr** [comment]
I don't really think it should be mandatory in the specs, because it would encompass all the Frictionless Libraries. As a quality of life feature, I would argue that this makes more sense in the Frictionless Frameworks.

I understand that allowing both is an extra burden for implementors and I would be happy to submit a PR. But I understand that the initial development is not at all the more relevant maintenance cost.

**Comment by peterdesmet** [comment]
Reopening.

Implementation notes:

- [ ] Add a dependency (`yaml`)
- [ ] Main functionality should be done in helper function `read_json()`
- [ ] Rename helper function to `read_json_yaml()`
- [ ] Select how to read file (`fromJSON` vs `read_yaml`) based on the extension: if not `.json`, try `read_yaml`. Don't test on `.yaml` because `.yml` is also often used.
- [ ] Update call to `read_json()` in `read_package()` and `get_schema()`
- [ ] Add tests and test files for both functions

**Comment by peterdesmet** [comment]
@fjuniorr This is now implemented on the main branch and will be included in version `0.10.0`. You can use it by reinstalling `devtools::install_github("frictionlessdata/frictionless-r")`.

The above example now gives:

``` r
library("frictionless")
dp <- read_package("https://raw.githubusercontent.com/dados-mg/datapackage-reprex/yaml-schema/datapackage.json")
#> Please make sure you have the right to access data from this Data Package for your intended use.
#> Follow applicable norms or requirements to credit the dataset and its authors.
df <- dp |> read_resource("estados")
df
#> # A tibble: 27 × 3
#>      COD NOME               SIGLA
#>    <dbl> <chr>              <chr>
#>  1    35 São Paulo          " SP"
#>  2    41 Paraná             " PR"
#>  3    42 Santa Catarina     " SC"
#>  4    43 Rio Grande do Sul  " RS"
#>  5    50 Mato Grosso do Sul " MS"
#>  6    11 Rondônia           " RO"
#>  7    12 Acre               " AC"
#>  8    13 Amazonas           " AM"
#>  9    14 Roraima            " RR"
#> 10    15 Pará               " PA"
#> # … with 17 more rows
```

<sup>Created on 2022-01-14 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by fjuniorr** [comment]
In our workflow this is going to be super useful!

Thanks very much!
